### PR TITLE
[v4] migrate roles page header

### PR DESCRIPTION
**Pull request by markkaylor**
### What does it do?

It migrates the Header component for the RolesList Page

### CODE REVIEW

File: packages/plugins/users-permissions/admin/src/pages/Roles/ListPage/index.js [hunk 1]
```diff
@@ -1,195 +1,59 @@
-import React, { useCallback, useMemo, useState } from 'react';
-import { List, Header } from '@buffetjs/custom';
-import { Helmet } from 'react-helmet';
+import React from 'react';
+import { Button, HeaderLayout, Layout, Main } from '@strapi/parts';
+import { AddIcon } from '@strapi/icons';
 import { useIntl } from 'react-intl';
+import { useTracking, SettingsPageTitle } from '@strapi/helper-plugin';
```
Comment (soupette) [inline packages/plugins/users-permissions/admin/src/pages/Roles/ListPage/index.js:5]: 👍

File: packages/plugins/users-permissions/admin/src/pages/Roles/ListPage/index.js [hunk 2]
```diff
@@ -1,195 +1,59 @@
...
-          e.preventDefault();
-          setModalDelete(role.id);
-          e.stopPropagation();
-        },
-      });
-    }
-
-    return links;
-  };
+  const pageTitle = formatMessage({
```
Comment (soupette) [inline packages/plugins/users-permissions/admin/src/pages/Roles/ListPage/index.js:21]: 👍 it's great that you thought about the defaultMessage here!

File: packages/plugins/users-permissions/admin/src/pages/Roles/ListPage/index.js [hunk 3]
```diff
@@ -1,195 +1,59 @@
...
+          title={formatMessage({
+            id: 'Settings.roles.title',
+            defaultMessage: 'Roles',
+          })}
+          subtitle={formatMessage({
+            id: 'Settings.roles.list.description',
+            defaultMessage: 'List of roles',
+          })}
+          primaryAction={
+            <Button onClick={handleNewRoleClick} startIcon={<AddIcon />}>
```
Comment (soupette) [inline packages/plugins/users-permissions/admin/src/pages/Roles/ListPage/index.js:47]: There is a permission that you need to apply here you can use the following component:


```
import { CheckPermissions } from '@strapi/helper-plugin';

primaryAction={<CheckPermissions permissions={permissions.createRole}><Button {...} /></CheckPermissions>}

```

**Review by soupette — CHANGES_REQUESTED** [review]
Good job!

File: packages/plugins/users-permissions/admin/src/pages/Roles/ListPage/tests/index.test.js [hunk 4]
```diff
@@ -0,0 +1,160 @@
...
+import { IntlProvider } from 'react-intl';
+import { ThemeProvider, lightTheme } from '@strapi/parts';
+
+import RoleListPage from '../index';
+
+jest.mock('@strapi/helper-plugin', () => ({
+  ...jest.requireActual('@strapi/helper-plugin'),
+  useTracking: jest.fn(() => ({ trackUsage: jest.fn() })),
+  useNotification: jest.fn(),
+  CheckPermissions: jest.fn(() => <div />),
```
Comment (markkaylor) [inline packages/plugins/users-permissions/admin/src/pages/Roles/ListPage/tests/index.test.js:12]: not sure I did this correctly...

**Review by soupette — APPROVED** [review]
LGTM

**Pull request merged**